test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsUnitTest.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.authentication;
+
+import net.sf.json.JSON;
+import org.junit.Before;
+import org.junit.Test;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.equalToIgnoringCase;
+import static org.hamcrest.core.AllOf.allOf;
+
+import org.zaproxy.zap.extension.api.ApiResponse;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * @author Vahid Rafiei (@vahid_r)
+ */
+public class UsernamePasswordAuthenticationCredentialsUnitTest {
+
+    private UsernamePasswordAuthenticationCredentials usernamePasswordAuthenticationCredentials;
+    private UsernamePasswordAuthenticationCredentials notConfiguredInstance;
+    private String username = "myUsername";
+    private String password = "myPassword";
+
+    @Before
+    public void setUp() {
+        this.usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+        this.notConfiguredInstance = new UsernamePasswordAuthenticationCredentials();
+    }
+
+    @Test
+    public void shouldBeConfiguredIfUsernameAndPasswordAreNotNull() throws Exception {
+        // Given/When
+        boolean isConfigured = usernamePasswordAuthenticationCredentials.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(true));
+    }
+
+    @Test
+    public void shouldNotBeConfiguredIfUsernameAndPasswordAreNull() throws Exception {
+        // Given/When
+        boolean isConfigured = notConfiguredInstance.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(false));
+    }
+
+
+    @Test
+    public void shouldEncodeMethodReturnNullPatternIfUsernameIsNull() throws Exception {
+        // Given
+        String nullPattern = "AA==";
+        username = null;
+        password = "something";
+        String stringSeparator = "|";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        String encodedResult = usernamePasswordAuthenticationCredentials.encode(stringSeparator);
+
+        // Then
+        assertThat(encodedResult, is(nullPattern));
+    }
+
+    @Test
+    public void shouldEncodeUsernameAndPasswordWithANonFieldSeparator() throws Exception {
+        // Given
+        List<String> sampleNonFieldSeparators = Arrays.asList("-", "|", "/", "\\", "+");
```

Comment:
I'm probably missing something subtle here but in the previous test the pipe character was specifically used, suggesting it's the correct separator. This test seems to suggest it's among the wrong choices....